test/api/model/initial_snapshot_test.dart

```diff
@@ -40,4 +40,24 @@ void main() {
       'unread_message_ids': [11, 2, 3],
     })).throws<AssertionError>();
   });
+
+  test('UnreadMessagesSnapshot from json recognizes channels as streams', () {
+    final snapshot = UnreadMessagesSnapshot.fromJson({
+      'count': 1,
+      'pms': <dynamic>[],
+      'huddles': <dynamic>[],
+      'mentions': <dynamic>[],
+      'old_unreads_missing': false,
+      'streams': [{
+        'stream_id': 1,
+        'topic': 'topic name',
+        'unread_message_ids': [1, 2]}]
+    });
+
+    check(snapshot.channels.length).equals(1);
+    check(snapshot.channels.first.toJson()).deepEquals(
+      UnreadChannelSnapshot(
+        topic: 'topic name', streamId: 1,
+        unreadMessageIds: [1, 2]).toJson());
```

Comment:
This can be simplified with our `jsonEquals`:
```suggestion
    check(snapshot.channels).single.jsonEquals(
      UnreadChannelSnapshot(
        topic: 'topic name', streamId: 1,
        unreadMessageIds: [1, 2]));
 ```